src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs

```diff
@@ -192,6 +192,11 @@ public virtual Task HandlePingAsync(PingMessage pingMessage)
                 Log.ReceivedServiceStatusPing(Logger, status, Endpoint);
                 _hasClients = status;
                 Endpoint.IsActive = GetServiceStatus(status, CheckWindow, CheckTimeSpan);
+
+                if (RuntimeServicePingMessage.TryGetClientCount(pingMessage, out var clientCount))
```

Comment:
is it intentional to have the check inside the status check?
PS: add ut

Reply:
Yes. Since the count number is together with status ping, and other pings are separated. Put inside would save some checks.